stores/utxo/src/main/java/com/bloxbean/cardano/yaci/store/utxo/storage/impl/UtxoStorageReaderImpl.java

```diff
@@ -54,6 +55,27 @@ public List<AddressUtxo> findUtxoByAddress(@NonNull String address, int page, in
                 .toList();
     }
 
+    @Override
+    public List<AddressUtxo> findUtxosByAsset(String unit, int page, int count, Order order) {
+        Pageable pageable = PageRequest.of(page, count)
+                .withSort(order.equals(Order.desc) ? Sort.Direction.DESC : Sort.Direction.ASC, "slot", "txHash", "outputIndex");
+
+        var query = dsl
+                .select(ADDRESS_UTXO.fields())
+                .from(ADDRESS_UTXO)
+                .leftJoin(TX_INPUT)
+                .using(field(ADDRESS_UTXO.TX_HASH), field(ADDRESS_UTXO.OUTPUT_INDEX))
+                .where(field(ADDRESS_UTXO.AMOUNTS).cast(String.class).contains(unit))
```

Comment:
@edridudi Fyi, We are currently using a workaround by checking assets through the contains method, but this may create some edge case scenarios, such as when there are two assets with the same policy and a similar prefix in their names.

I understand that other queries in this class are following the same approach. However, I think we should plan to change this query to utilize JOOQ's JSON query support. We may want to look into that in this PR or we can plan it in a separate PR.

Reply:
changed this to equalIgnoreCase